scripts/test/Muon/fft_presenter_context_interaction_test.py

```diff
@@ -0,0 +1,149 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+import unittest
+from Muon.GUI.Common import mock_widget
+from qtpy import QtWidgets
+from mantid.py3compat import mock
+from Muon.GUI.Common.utilities import load_utils
+from Muon.GUI.Common import thread_model
+from Muon.GUI.FrequencyDomainAnalysis.FFT import fft_presenter_new
+from Muon.GUI.FrequencyDomainAnalysis.FFT import fft_view
+from Muon.GUI.FrequencyDomainAnalysis.FFT import fft_model
+from Muon.GUI.FrequencyDomainAnalysis.frequency_context import FrequencyContext
+from Muon.GUI.Common.contexts.context_setup import setup_context_for_tests
+from Muon.GUI.Common.muon_pair import MuonPair
+from mantid.api import FileFinder
+
+
+class FFTPresenterTest(unittest.TestCase):
+    def setUp(self):
+        self._qapp = mock_widget.mockQapp()
+        # Store an empty widget to parent all the views, and ensure they are deleted correctly
+        self.obj = QtWidgets.QWidget()
+        setup_context_for_tests(self)
+
+        self.data_context.instrument = 'MUSR'
+        self.frequency_context = FrequencyContext(self.context)
+
+        self.gui_context.update({'RebinType': 'None'})
+
+        self.view = fft_view.FFTView(self.obj)
+        self.model1 = fft_model.FFTModel()
+        self.model = fft_model.FFTWrapper
+
+        self.presenter = fft_presenter_new.FFTPresenter(
+            self.view, self.model, self.context)
+
+        file_path = FileFinder.findRuns('MUSR00022725.nxs')[0]
+        ws, run, filename = load_utils.load_workspace_from_filename(file_path)
+        self.data_context._loaded_data.remove_data(run=run)
+        self.data_context._loaded_data.add_data(run=[run], workspace=ws, filename=filename, instrument='MUSR')
+        self.data_context.current_runs = [[22725]]
+
+        self.context.update_current_data()
+        test_pair = MuonPair('test_pair', 'top', 'bottom', alpha=0.75)
+        self.group_context.add_pair(pair=test_pair)
+
+        self.view.warning_popup = mock.MagicMock()
+
+    def test_getWorkspaceNames_sets_workspace_and_imaginary_workspace_list_correctly(self):
+        self.presenter.getWorkspaceNames()
+
```

Comment:
lets check the length of the the two arrays to make sure they have the number of entries we think